pkg/codegen/go/gen.go

```diff
@@ -206,9 +206,16 @@ func (pkg *pkgContext) tokenToType(tok string) string {
 	if mod == "" {
 		mod = packageRoot(pkg.pkg)
 	}
-	mod = strings.ReplaceAll(mod, "/", "")
-	mod = strings.ReplaceAll(mod, "-", "") + "." + name
-	return strings.ReplaceAll(mod, "-provider", "")
+
+	var importPath string
+	if alias, hasAlias := pkg.pkgImportAliases[path.Join(pkg.importBasePath, mod)]; hasAlias {
+		importPath = alias
+	} else {
+		importPath = strings.ReplaceAll(mod, "/", "")
+		importPath = strings.ReplaceAll(importPath, "-", "")
+	}
+
+	return strings.ReplaceAll(importPath+"."+name, "-provider", "")
```

Comment:
It would be great to add some unit tests for these two functions while you are on it

Reply:
Done